src/ssl/support.cc

```diff
@@ -962,14 +962,16 @@ void
 Ssl::chainCertificatesToSSLContext(Security::ContextPointer &ctx, Security::ServerOptions &options)
 {
     assert(ctx);
-    // Add signing certificate to the certificates chain
-    X509 *signingCert = options.signingCa.cert.get();
-    if (SSL_CTX_add_extra_chain_cert(ctx.get(), signingCert)) {
-        // increase the certificate lock
-        X509_up_ref(signingCert);
-    } else {
-        const auto ssl_error = ERR_get_error();
-        debugs(33, DBG_IMPORTANT, "WARNING: can not add signing certificate to SSL context chain: " << Security::ErrorString(ssl_error));
+    if (!options.signingCa.selfSigned) {
+        // Add signing certificate to the certificates chain
+        X509 *signingCert = options.signingCa.cert.get();
+        if (SSL_CTX_add_extra_chain_cert(ctx.get(), signingCert)) {
+            // increase the certificate lock
+            X509_up_ref(signingCert);
+        } else {
+            const auto ssl_error = ERR_get_error();
+            debugs(33, DBG_IMPORTANT, "WARNING: can not add signing certificate to SSL context chain: " << Security::ErrorString(ssl_error));
+        }
     }
 
     for (auto cert : options.signingCa.chain) {
```

Comment:
Please add a C++ comment to explain why this loop should run when options.signingCa.selfSigned is true.

Reply:
The loop never runs because the options.signingCa.chain is always empty when options.signingCa.selfSigned is true.
I fixed to just return from this function if options.signingCa.selfSigned is true.

commit f6a62e8f66